tests/processing/test_block_bootstrap.py

```diff
@@ -0,0 +1,306 @@
+            {"dim2": 2, "dim3": 2},
+            3,
+            [["dim1"]],
+            True,
+            (10, 5, 7, 3),
+            xr.DataArray,
+        ),
+        # Dataset bootstrap
+        (
+            [
+                xr.Dataset(
+                    {
+                        "var1": xr.DataArray(np.random.rand(10, 5), dims=["dim1", "dim2"]),
+                        "var2": xr.DataArray(np.random.rand(10, 5), dims=["dim1", "dim2"]),
+                    }
+                )
+            ],
+            {"dim1": 2, "dim2": 2},
+            3,
+            None,
+            True,
+            (10, 5, 3),
+            xr.Dataset,
+        ),
+    ],
+)
+def test_block_bootstrap(objects, blocks, n_iteration, exclude_dims, circular, expected_shape, expected_type):
+    """Test block_bootstrap works as expected"""
+    result = block_bootstrap(
+        *objects, blocks=blocks, n_iteration=n_iteration, exclude_dims=exclude_dims, circular=circular
+    )
+    if expected_type == tuple:
+        assert isinstance(result, tuple)
+        assert all(isinstance(res, xr.DataArray) for res in result)
+        for res in result:
+            assert res.shape == expected_shape
+    elif expected_type == xr.Dataset:
+        assert isinstance(result, xr.Dataset)
+        for var in result.data_vars:
+            assert result[var].shape == expected_shape
+    else:
+        assert isinstance(result, expected_type)
+        assert result.shape == expected_shape
+
+
+@pytest.mark.parametrize(
+    "objects, blocks, n_iteration, exclude_dims, circular, expected_shape",
+    [
+        (
+            [xr.DataArray(np.random.rand(10, 5), dims=["dim1", "dim2"]).chunk()],
+            {"dim1": 2, "dim2": 2},
+            3,
+            None,
+            True,
+            (10, 5, 3),
+        ),
+        # Dask arrays to meet block_size < 1
+        (
+            [xr.DataArray(da.random.random((1000, 1000, 30), chunks=dict(dim1=-1)), dims=["dim1", "dim2", "dim3"])],
+            {"dim1": 2, "dim2": 2},
+            2,
+            None,
+            True,
+            (30, 1000, 1000, 2),
+        ),
+        # Dask arrays for a case with leftover != 0
+        (
+            [xr.DataArray(da.random.random((1000, 1000, 10), chunks=dict(dim1=-1)), dims=["dim1", "dim2", "dim3"])],
+            {"dim1": 2, "dim2": 2},
+            3,
+            None,
+            True,
+            (10, 1000, 1000, 3),
+        ),
+    ],
+)
+def test_block_bootstrap_dask(objects, blocks, n_iteration, exclude_dims, circular, expected_shape):
+    """Test block_bootstrap can work with dask arrays"""
+    if dask == "Unavailable":  # pragma: no cover
+        pytest.skip("Dask unavailable, could not run test")  # pragma: no cover
+
+    result = block_bootstrap(
+        *objects, blocks=blocks, n_iteration=n_iteration, exclude_dims=exclude_dims, circular=circular
+    )
+    assert result.shape == expected_shape
```

Comment:
Can you test that that `result.data` is a dask object? Then when you run `result.compute()` the underlying object is an `np.ndarray`? See https://github.com/nci/scores/blob/develop/tests/categorical/test_multicategorical.py#L283 for an example.